sympy/matrices/expressions/tests/test_derivatives.py

```diff
@@ -288,6 +289,30 @@ def test_mixed_deriv_mixed_expressions():
     assert expr.diff(A) == (3*Trace(A)**2)*Identity(k)
 
 
+def test_derivatives_matrix_norms():
+
+    expr = x.T*y
+    assert expr.diff(x) == y
+
+    expr = (x.T*y)**S.Half
+    assert expr.diff(x) == y/(2*sqrt(x.T*y))
+
+    expr = (x.T*x)**S.Half
+    assert expr.diff(x) == x*(x.T*x)**S.Half
```

Comment:
```suggestion
    assert expr.diff(x) == x*(x.T*x)**(-S.Half)
```
It's supposed to be `x*(x.T*x)**(-1/2)`